packages/schematics/src/add/index.ts

```diff
@@ -28,180 +30,138 @@ if (fs.existsSync(join(__dirname, '../../package.json'))) {
+  return (host: Tree) =>
+    addModuleImportToModule(
+      host,
+      'BrowserAnimationsModule',
+      '@angular/platform-browser/animations',
+      projectSettings.module
+    );
+}
+
+function addClarityDependencies(host: Tree) {
+  updateJsonFile(host, 'package.json', json => {
+    const version = getVersion(json.dependencies['@angular/core'], corePackage.version);
+
+    json.dependencies['@clr/angular'] = `${version}`;
+    json.dependencies['@clr/ui'] = `${version}`;
+    json.dependencies['@clr/icons'] = `${version}`;
+
+    const packages = Object.keys(json.dependencies);
+    if (!packages.includes('@webcomponents/webcomponentsjs')) {
+      json.dependencies['@webcomponents/webcomponentsjs'] = '^2.0.0';
+    }
+  });
 }
 
 // Checks if a version of Angular is compatible with current or next
 function getVersion(ngVersion: string, clrVersion: string) {
   const diff = 6; // Number disparity between Angular and Clarity, this works as long as we stay in sync with versioning
-  const version1 = Number.parseInt(ngVersion.split('.')[0].replace(/\D/g, ''));
-  const version2 = Number.parseInt(clrVersion.split('.')[0].replace(/\D/g, ''));
+  const majorNgVersion = Number.parseInt(ngVersion.split('.')[0].replace(/\D/g, ''));
+  const majorClrVersion = Number.parseInt(clrVersion.split('.')[0].replace(/\D/g, ''));
 
-  if (version1 - version2 < diff) {
+  if (majorNgVersion - majorClrVersion < diff) {
     // If Angular is less than 6 versions ahead, backtrack Clarity version
-    return `^${version1 - diff}.0.0`;
+    return `^${majorNgVersion - diff}.0.0`;
   } else {
     // Otherwise, just link to installed version for latest or next releases
     return clrVersion;
   }
 }
 
-// Handles adding a module to the NgModule
-function addDeclarationToNgModule(
-  options: ComponentOptions,
-  imports: string,
-  moduleName: string,
-  packageName: string
-): Rule {
-  return (host: Tree) => {
-    const modulePath = options.module as string;
-
-    const text = host.read(modulePath);
-    if (text === null) {
-      throw new SchematicsException(`File ${modulePath} does not exist.`);
-    }
-    const sourceText = text.toString('utf-8');
-    const source = ts.createSourceFile(modulePath, sourceText, ts.ScriptTarget.Latest, true);
-
-    const declarationChanges = addSymbolToNgModuleMetadata(source, modulePath, imports, moduleName, packageName);
+function addAssetsToConfigFile(host: Tree) {
+  updateJsonFile(host, CONFIG_FILE_NAME, json => {
+    const projects = Object.keys(json.projects);
+    const project = projects.find(key => key === projectSettings.project);
 
-    const declarationRecorder = host.beginUpdate(modulePath);
-    for (const change of declarationChanges) {
-      if (change instanceof InsertChange) {
-        declarationRecorder.insertLeft(change.pos, change.toAdd);
-      }
+    if (!project) {
+      console.info(`Could not update CLI config file to add scripts and styles. You'll have to add them manually.`);
```

Comment:
Why don't we use `SchematicsException` here?

Seems that this the end of the road if the project is missing? Or this is the happy path and this is just to say that "OK nothing to do more here"?

Not sure is the change required but it seems like a `info` may look to someone that "the task had done some work and finished successfully "

Reply:
Maybe `warn` is a better method here? Throwing an exception seems a little to heavey here unless we consider this an error that angular.json arrays could not be updated.

Reply:
I'm ok with a `warn` just the message seems more like "Oops, this task could not finish successfully" .

But `warn` is fine ...

Reply:
I'll switch the `info` for `warn`.